pkg/controller/elasticsearch/user/roles.go

```diff
@@ -150,6 +152,22 @@ var (
 				},
 			},
 		},
+		LogstashUserRole: esclient.Role{
```

Comment:
> The indices chosen here will restrict logstash to only being able to create the "standard" indices/templates, and not custom ones created by users.

That's a good point. New roles can be added using [the `roles` section](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-users-and-roles.html#k8s_creating_custom_roles), and even if it is not ideal this can also be used to override predefined roles.

Reply:
I feel like the uncommented/explained use of `superuser` roles by Agent and Enterprise Search is actually somewhat of a bug. Even if there is a motivation for it in order to create and ingest data into custom indices, it feels like something that should be called out in a comment/doc at the very least and probably a custom role appropriate privileges to create arbitrary indices would have still been better than `superuser`.